### PR TITLE
Update tarantool

### PR DESCRIPTION
**Pull request by headshog**

### CODE REVIEW

File: projects/tarantool/Dockerfile [hunk 1]
```diff
@@ -47,3 +47,8 @@ COPY build.sh main.c afl.cc test/fuzz/
 RUN test/fuzz/build.sh
 
 WORKDIR /
+
+# Unzip new corpus directories
+COPY corpus_datetime_parse_full.zip corpus_datetime_strptime.zip corpus_mp_datetime.zip /
```
Comment (kobrineli) [inline projects/tarantool/Dockerfile:52]: get these corporas from https://github.com/ligurio/tarantool-corpus

File: projects/tarantool/build.sh [hunk 2]
```diff
@@ -120,8 +120,6 @@ cmake_args=(
 )
 [[ -e build ]] && rm -rf build
 mkdir -p build/test/fuzz
-export LIB_FUZZING_ENGINE="$PWD/build/test/fuzz/afl.o"
-$CC $CFLAGS -c test/fuzz/afl.cc -o $LIB_FUZZING_ENGINE
```
Comment (kobrineli) [inline projects/tarantool/build.sh:124]: remove `afl.cc` from repo

**Review by kobrineli — CHANGES_REQUESTED** [review]

**Review by kobrineli — COMMENTED** [review]
Add cmplog instrumentation

File: projects/tarantool/datetime_strptime-afl++.toml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+jobs = 2
+
+[[aflplusplus]]
+args = "-t 60000 -i /datetime_strptime"
+target = "/datetime_strptime_afl @@"
+
+[[aflplusplus]]
+args = "-c /datetime_strptime_cmplog -t 60000 -i /datetime_strptime"
+target = "/datetime_strptime_afl @@"
+
```
Comment (kobrineli) [inline projects/tarantool/datetime_strptime-afl++.toml:29]: extra line

**Review by kobrineli — CHANGES_REQUESTED** [review]
Remove *.zip files

Update README

File: projects/tarantool/Dockerfile [hunk 4]
```diff
@@ -36,14 +36,21 @@ RUN git clone https://github.com/tarantool/tarantool
...
 RUN test/fuzz/build.sh
 
 WORKDIR /
+
+# Get new corpus directories
+RUN git clone https://github.com/ligurio/tarantool-corpus && cd /tarantool-corpus && \
+    git checkout dc6a7b9c663df970d1071a1a40881b30c5f54d57 && cp -r mp_datetime/ datetime_strptime/ datetime_parse_full/ /
+
+# Fix uri corpus directory
+RUN rm /corpus_uri/adc83b19e793491b1c6ea0fd8b46cd9f32e592fc
```
Comment (anfedotoff) [inline projects/tarantool/Dockerfile:56]: No new line at end of file

File: projects/tarantool/Dockerfile [hunk 5]
```diff
@@ -36,14 +36,21 @@ RUN git clone https://github.com/tarantool/tarantool
...
+COPY build.sh test/fuzz/
 
 # Build fuzz targets.
 RUN test/fuzz/build.sh
 
 WORKDIR /
+
+# Get new corpus directories
+RUN git clone https://github.com/ligurio/tarantool-corpus && cd /tarantool-corpus && \
+    git checkout dc6a7b9c663df970d1071a1a40881b30c5f54d57 && cp -r mp_datetime/ datetime_strptime/ datetime_parse_full/ /
```
Comment (SweetVishnya) [inline projects/tarantool/Dockerfile:53]: Maybe move corpuses and remove git repo?

File: projects/tarantool/datetime_parse_full-afl++.toml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+[[aflplusplus]]
+args = "-t 60000 -i /datetime_parse_full"
+target = "/datetime_parse_full_afl @@"
+
+[[aflplusplus]]
+args = "-c /datetime_parse_full_cmplog -t 60000 -i /datetime_parse_full"
+target = "/datetime_parse_full_afl @@"
+
+[cov]
+target = "/datetime_parse_full_cov @@"
```
Comment (anfedotoff) [inline projects/tarantool/datetime_parse_full-afl++.toml:31]: new line

File: projects/tarantool/datetime_parse_full.toml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+args = "-l debug -j 2"
+target = "/datetime_parse_full_sydr @@"
+jobs = 2
+
+[libfuzzer]
+path = "/datetime_parse_full_fuzzer"
+args = "/datetime_parse_full"
+
+[cov]
+target = "/datetime_parse_full_cov @@"
```
Comment (anfedotoff) [inline projects/tarantool/datetime_parse_full.toml:27]: new line

File: projects/tarantool/datetime_strptime.toml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+args = "-l debug -j 2"
+target = "/datetime_strptime_sydr @@"
+jobs = 2
+
+[libfuzzer]
+path = "/datetime_strptime_fuzzer"
+args = "/datetime_strptime"
+
+[cov]
+target = "/datetime_strptime_cov @@"
```
Comment (anfedotoff) [inline projects/tarantool/datetime_strptime.toml:27]: new line

File: projects/tarantool/csv-afl++.toml [hunk 9]
```diff
@@ -19,9 +19,13 @@ args = "-l debug -j 2"
...
 jobs = 2
 
-[aflplusplus]
+[[aflplusplus]]
 args = "-t 60000 -i /corpus_csv"
-target = "/csv_afl"
+target = "/csv_afl @@"
+
+[[aflplusplus]]
+args = "-c /csv_cmplog -t 60000 -i /corpus_csv"
```
Comment (SweetVishnya) [inline projects/tarantool/csv-afl++.toml:27]: -m none -l AT

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: projects/tarantool/README.md [hunk 10]
```diff
@@ -37,7 +37,7 @@ Collect and report coverage:
 
 ## Alternative Fuzz Targets
 
-Tarantool project has 5 fuzz targets.
+Tarantool project has 7 fuzz targets.
```
Comment (kobrineli) [inline projects/tarantool/README.md:40]: 8?

File: projects/tarantool/csv-afl++.toml (hunk 9 above)
Comment (headshog) [reply projects/tarantool/csv-afl++.toml:27]: In every afl++.toml file?

**Review by anfedotoff — APPROVED** [review]

**Review by kobrineli — APPROVED** [review]

File: projects/tarantool/Dockerfile [hunk 11]
```diff
@@ -36,14 +36,21 @@ RUN git clone https://github.com/tarantool/tarantool
...
+COPY build.sh test/fuzz/
 
 # Build fuzz targets.
 RUN test/fuzz/build.sh
 
 WORKDIR /
+
+# Get new corpus directories
+RUN git clone https://github.com/ligurio/tarantool-corpus && cd /tarantool-corpus && \
+    git checkout dc6a7b9c663df970d1071a1a40881b30c5f54d57 && mv mp_datetime/ datetime_strptime/ datetime_parse_full/ && rm -rf tarantool-corpus
```
Comment (SweetVishnya) [inline projects/tarantool/Dockerfile:53]: cd .. before rm

Comment (SweetVishnya) [reply projects/tarantool/Dockerfile:53]: mv destination

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: projects/tarantool/build.sh [hunk 12]
```diff
@@ -132,6 +130,44 @@ do
...
+# Building cmplog instrumentation.
+export AFL_LLVM_CMPLOG=1
+cmake_args=(
+    # Specific to Tarantool
+    -DENABLE_BACKTRACE=OFF
+    -DENABLE_FUZZER=ON
+    -DOSS_FUZZ=OFF
+    -DENABLE_ASAN=ON
+    -DENABLE_UB_SANITIZER=ON
+    -DCMAKE_BUILD_TYPE=Release
```
[the remark this answers is not in the document]
Comment (headshog) [reply projects/tarantool/build.sh]: Done

**Review by SweetVishnya — APPROVED** [review]
LGTM

**Pull request merged**